torchao/experimental/ops/mps/test/test_quantizer.py

```diff
@@ -0,0 +1,104 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+import copy
+import os
+import sys
+
+import torch
+import torchao_mps_ops
+import unittest
+
+torchao_root: Optional[str] = os.getenv("TORCHAO_ROOT")
+assert torchao_root is not None, "TORCHAO_ROOT is not set"
+
+sys.path.insert(0, torchao_root)
+from torchao.experimental.quant_api import IntxWeightOnlyLinearQuantizer
+
+
+def parameterized(test_cases):
+    def decorator(func):
+        def wrapper(self):
+            for case in test_cases:
+                with self.subTest(case=case):
+                    func(self, *case)
+        return wrapper
+    return decorator
+
+
+class TestIntxWeightOnlyLinearQuantizer(unittest.TestCase):
+    cases = [(nbit,) for nbit in range(1, 8)]
+
+    def _model_setup(self):
+        k0 = 512
+        k1 = 256
+        k2 = 128
+        k3 = 1024
```

Comment:
Check for non power of 2 and odd/prime number sizes. If they are not supposed to work than have expect failure tests.

Reply:
I updated the values, to cover non-power of 2 values. But right now, the quantization code inside quant_api only supports values of K that are a multiple of group_size. I added a TODO, to address this in a future diff. I don't want to make many changes to the quant_api code in this diff, since that is also used by the lowbit cpu kernels